client/app/src/main/java/com/microsoft/research/karya/ui/registration/PhoneNumberFragment.kt

```diff
@@ -1,29 +1,59 @@
-// Copyright (c) Microsoft Corporation.
-// Licensed under the MIT license.
-
 package com.microsoft.research.karya.ui.registration
 
 import android.content.Intent
 import android.os.Bundle
 import android.text.Editable
 import android.text.TextWatcher
 import android.util.TypedValue
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
 import android.view.View
+import android.view.ViewGroup
 import com.microsoft.research.karya.R
 import com.microsoft.research.karya.ui.base.BaseActivity
-import kotlinx.android.synthetic.main.activity_phone_number.*
+import kotlinx.android.synthetic.main.fragment_phone_number.*
+import kotlinx.android.synthetic.main.fragment_phone_number.view.*
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
 
-private const val PHONE_NUMBER_LENGTH = 10
+class PhoneNumberFragment : Fragment() {
 
-class PhoneNumberActivity : BaseActivity(useAssistant = true) {
+    private val PHONE_NUMBER_LENGTH = 10
+    private lateinit var registrationActivity: RegistrationActivity
+    private lateinit var baseActivity: BaseActivity
 
-    /** Android strings */
-    private var phoneNumberPromptMessage: String = ""
+    protected val ioScope = CoroutineScope(Dispatchers.IO)
+    protected val uiScope = CoroutineScope(Dispatchers.Main)
 
     override fun onCreate(savedInstanceState: Bundle?) {
-        setContentView(R.layout.activity_phone_number)
         super.onCreate(savedInstanceState)
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View? {
+
+        registrationActivity = activity as RegistrationActivity
+        baseActivity = activity as BaseActivity
```

Comment:
Let's not keep dependencies over `BaseActivity`.

Reply:
Its not a dependency. We are just using the activity instance as BaseActivity since, it inherits from it but anyways, we are removing this once we remove BaseActivity entirely